pkg/liqoctl/output/output.go

```diff
@@ -98,6 +98,11 @@ type Printer struct {
 	verbose    bool
 }
 
+func (p *Printer) SpinnerPrintWhileRunning(spinner *pterm.SpinnerPrinter, f func(message ...interface{}), message ...interface{}) *pterm.SpinnerPrinter {
+	f(message...)
+	return p.StartSpinner(spinner.Text)
+}
```

Comment:
```suggestion
func (p *Printer) SpinnerPrintWhileRunning(spinner *pterm.SpinnerPrinter, f func()) *pterm.SpinnerPrinter {
	f()
	return p.StartSpinner(spinner.Text)
}
```

and then in the caller

```
s = o.Printer.SpinnerPrintWhileRunning(s, func() {s.Warning(warning)})
```

all these interfaces and the parameter not used but forwarded to the function seams strange and dangerous

Reply:
Otherwise, we can create a dedicated function for  Warning and Success, which take only the spinner and the  message as arguments.